### ISSUE TITLE
Dark oak processed incorrectly?

### ISSUE THREAD

**Issue by molor** [slipcor/TreeAssist#75]
Hello. 

I spent 1 hour and could not achieve that such trees
![2021-06-24_12 47 14](https://user-images.githubusercontent.com/19989717/123242489-be97fe80-d4ea-11eb-980a-a0d3ee3bb2a0.png)

were break completely, without leaving this blocks on the sides
![2021-06-24_12 47 19](https://user-images.githubusercontent.com/19989717/123242550-cd7eb100-d4ea-11eb-828f-9ec33d5598ad.png)

I tried many values in `Radius`, `Height`, `Thickness` etc of every section (in `default.yml` and `dark_oak.yml`), tried adding materials to almost every list but this doesn't change anything. Could you please share settings with me that can solve this?

**Comment by molor** [comment]
It seems that tall (not dark) oak branches are also not processed normally

**Comment by slipcor** [comment]
I will look into this. Which biome is this? Last time I went through a dark oak forest / roofed forest I think the trees worked pretty well.

The tall tree is sort of known to be very finicky. I will also look into this. I recently added a tool to grow a tree for debug purposes, I will enhance this debugging function. I hope you can work with the current state until then, as currently I have no idea what to do about this.

If you have a lot of trees and you know one of them that will fail, I would appreciate a debug there.

/treeassist debug all

try killing the tree

/treeassist debug off

and if the tree fails ,send me the contents of /plugins/treeassist/debug

It might be helpful, for me at least, if you could at least temporarily join my discord. We can then do this together, in real time. The discord has a new build for ticket #73 :)

**Comment by molor** [comment]
It's `dark_forest`

```
21.06.24 19:19:45 [INFO] 532 --- checking config tall_spruce.yml
21.06.24 19:19:45 [INFO] 533 checking for material BEDROCK
21.06.24 19:19:45 [INFO] 533 checking for material SPRUCE_LOG
21.06.24 19:19:45 [INFO] 533 --- checking config spruce.yml
21.06.24 19:19:45 [INFO] 533 checking for material BEDROCK
21.06.24 19:19:45 [INFO] 533 checking for material SPRUCE_LOG
21.06.24 19:19:45 [INFO] 533 --- checking config mushroom-brown.yml
21.06.24 19:19:45 [INFO] 533 checking for material BEDROCK
21.06.24 19:19:45 [INFO] 533 checking for material MUSHROOM_STEM
21.06.24 19:19:45 [INFO] 534 --- checking config mushroom-red.yml
21.06.24 19:19:45 [INFO] 534 checking for material BEDROCK
21.06.24 19:19:45 [INFO] 534 checking for material MUSHROOM_STEM
21.06.24 19:19:45 [INFO] 534 --- checking config mushroom.yml
21.06.24 19:19:45 [INFO] 534 checking for material BEDROCK
21.06.24 19:19:45 [INFO] 534 checking for material MUSHROOM_STEM
21.06.24 19:19:45 [INFO] 534 --- checking config tall_jungle.yml
21.06.24 19:19:45 [INFO] 534 checking for material BEDROCK
21.06.24 19:19:45 [INFO] 534 checking for material JUNGLE_LOG
21.06.24 19:19:45 [INFO] 534 --- checking config jungle.yml
21.06.24 19:19:45 [INFO] 534 checking for material BEDROCK
21.06.24 19:19:45 [INFO] 534 checking for material JUNGLE_LOG
21.06.24 19:19:45 [INFO] 534 --- checking config dark_oak.yml
21.06.24 19:19:45 [INFO] 534 checking for material BEDROCK
21.06.24 19:19:45 [INFO] 534 checking for material DARK_OAK_LOG
21.06.24 19:19:45 [INFO] 534 We hit the ground!
21.06.24 19:19:45 [INFO] 534 Tree found for Material DARK_OAK_LOG
21.06.24 19:19:45 [INFO] 534 Thickness: 4
21.06.24 19:19:45 [INFO] 534 continuing SOUTH
21.06.24 19:19:45 [INFO] 534 branching out
21.06.24 19:19:45 [INFO] 534 continuing EAST
21.06.24 19:19:45 [INFO] 535 continuing EAST
21.06.24 19:19:45 [INFO] 535 branching out
21.06.24 19:19:45 [INFO] 535 Trunks: 4
21.06.24 19:19:45 [INFO] 535 We hit the roof!
21.06.24 19:19:45 [INFO] 535 We hit the roof!
21.06.24 19:19:45 [INFO] 535 We hit the roof!
21.06.24 19:19:45 [INFO] 535 We hit the roof!
21.06.24 19:19:45 [INFO] 535 No other trunks found, checking again!
21.06.24 19:19:45 [INFO] 535 We hit the ground!
21.06.24 19:19:45 [INFO] 535 Thickness: 4
21.06.24 19:19:45 [INFO] 535 Not enough trunks found: 1
21.06.24 19:19:45 [INFO] 536 total checks: 176
21.06.24 19:19:45 [INFO] 536 Tree matches DARK_OAK_LOG
21.06.24 19:19:45 [INFO] 537 in hand: WOODEN_AXE
21.06.24 19:19:45 [INFO] 537 cooldown bypass permission!
21.06.24 19:19:45 [INFO] 537 in hand: WOODEN_AXE
21.06.24 19:19:45 [INFO] 537 success!
21.06.24 19:19:45 [INFO] 537 replanting is disabled!
21.06.24 19:19:45 [INFO] 537 Removing The Tree!
21.06.24 19:19:45 [INFO] 537 delay: 0; offset: 0
21.06.24 19:19:45 [INFO] 538 cooldown bypass permission!
21.06.24 19:19:45 [INFO] 538 pickup: false; mine: false
21.06.24 19:19:45 [INFO] 540 InstantRunner: 2b sadness:339/63/78 - DARK_OAK_LOG
21.06.24 19:19:45 [INFO] 540 breaking DARK_OAK_LOG. custom drops: false - roll: 0.9144235935552966
21.06.24 19:19:45 [INFO] 540 mat: DARK_OAK_LOG
21.06.24 19:19:45 [INFO] 540 not auto adding!
21.06.24 19:19:45 [INFO] 540 simply breaking sadness:339/63/78 - DARK_OAK_LOG
21.06.24 19:19:45 [INFO] 540 tool used: WOODEN_AXE
21.06.24 19:19:45 [INFO] 541 skipping damage for other reasons
21.06.24 19:19:45 [INFO] 551 InstantRunner: 2b sadness:339/63/79 - DARK_OAK_LOG
21.06.24 19:19:45 [INFO] 551 breaking DARK_OAK_LOG. custom drops: false - roll: 0.17711201799483034
21.06.24 19:19:45 [INFO] 551 mat: DARK_OAK_LOG
21.06.24 19:19:45 [INFO] 551 not auto adding!
21.06.24 19:19:45 [INFO] 551 simply breaking sadness:339/63/79 - DARK_OAK_LOG
21.06.24 19:19:45 [INFO] 551 tool used: WOODEN_AXE
21.06.24 19:19:45 [INFO] 552 skipping damage for other reasons
21.06.24 19:19:45 [INFO] 594 InstantRunner: 2b sadness:340/63/78 - DARK_OAK_LOG
21.06.24 19:19:45 [INFO] 594 breaking DARK_OAK_LOG. custom drops: false - roll: 0.023818562067552618
21.06.24 19:19:45 [INFO] 594 mat: DARK_OAK_LOG
21.06.24 19:19:45 [INFO] 594 not auto adding!
21.06.24 19:19:45 [INFO] 594 simply breaking sadness:340/63/78 - DARK_OAK_LOG
21.06.24 19:19:45 [INFO] 594 tool used: WOODEN_AXE
21.06.24 19:19:45 [INFO] 595 skipping damage for other reasons
21.06.24 19:19:45 [INFO] 644 InstantRunner: 2b sadness:340/63/79 - DARK_OAK_LOG
21.06.24 19:19:45 [INFO] 644 breaking DARK_OAK_LOG. custom drops: false - roll: 0.7588732974696785
21.06.24 19:19:45 [INFO] 644 mat: DARK_OAK_LOG
21.06.24 19:19:45 [INFO] 644 not auto adding!
21.06.24 19:19:45 [INFO] 644 simply breaking sadness:340/63/79 - DARK_OAK_LOG
21.06.24 19:19:45 [INFO] 644 tool used: WOODEN_AXE
21.06.24 19:19:45 [INFO] 645 skipping damage for other reasons
21.06.24 19:19:45 [INFO] 693 InstantRunner: 2b sadness:339/64/78 - DARK_OAK_LOG
21.06.24 19:19:45 [INFO] 693 breaking DARK_OAK_LOG. custom drops: false - roll: 0.2084960031969928
21.06.24 19:19:45 [INFO] 693 mat: DARK_OAK_LOG
21.06.24 19:19:45 [INFO] 693 not auto adding!
21.06.24 19:19:45 [INFO] 693 simply breaking sadness:339/64/78 - DARK_OAK_LOG
21.06.24 19:19:45 [INFO] 693 tool used: WOODEN_AXE
21.06.24 19:19:45 [INFO] 694 skipping damage for other reasons
21.06.24 19:19:45 [INFO] 743 InstantRunner: 2b sadness:339/65/79 - DARK_OAK_LOG
21.06.24 19:19:45 [INFO] 743 breaking DARK_OAK_LOG. custom drops: false - roll: 0.3179051651315622
21.06.24 19:19:45 [INFO] 743 mat: DARK_OAK_LOG
21.06.24 19:19:45 [INFO] 743 not auto adding!
21.06.24 19:19:45 [INFO] 743 simply breaking sadness:339/65/79 - DARK_OAK_LOG
21.06.24 19:19:45 [INFO] 743 tool used: WOODEN_AXE
21.06.24 19:19:45 [INFO] 744 skipping damage for other reasons
21.06.24 19:19:45 [INFO] 794 InstantRunner: 2b sadness:340/64/78 - DARK_OAK_LOG
21.06.24 19:19:45 [INFO] 794 breaking DARK_OAK_LOG. custom drops: false - roll: 0.04093075590958173
21.06.24 19:19:45 [INFO] 794 mat: DARK_OAK_LOG
21.06.24 19:19:45 [INFO] 794 not auto adding!
21.06.24 19:19:45 [INFO] 794 simply breaking sadness:340/64/78 - DARK_OAK_LOG
21.06.24 19:19:45 [INFO] 794 tool used: WOODEN_AXE
21.06.24 19:19:45 [INFO] 795 skipping damage for other reasons
21.06.24 19:19:45 [INFO] 844 InstantRunner: 2b sadness:340/67/78 - DARK_OAK_LOG
21.06.24 19:19:45 [INFO] 844 breaking DARK_OAK_LOG. custom drops: false - roll: 0.9857032716465929
21.06.24 19:19:45 [INFO] 844 mat: DARK_OAK_LOG
21.06.24 19:19:45 [INFO] 844 not auto adding!
21.06.24 19:19:45 [INFO] 844 simply breaking sadness:340/67/78 - DARK_OAK_LOG
21.06.24 19:19:45 [INFO] 844 tool used: WOODEN_AXE
21.06.24 19:19:45 [INFO] 845 skipping damage for other reasons
21.06.24 19:19:45 [INFO] 893 InstantRunner: 2b sadness:339/66/78 - DARK_OAK_LOG
21.06.24 19:19:45 [INFO] 893 breaking DARK_OAK_LOG. custom drops: false - roll: 0.5377548234242033
21.06.24 19:19:45 [INFO] 893 mat: DARK_OAK_LOG
21.06.24 19:19:45 [INFO] 893 not auto adding!
21.06.24 19:19:45 [INFO] 893 simply breaking sadness:339/66/78 - DARK_OAK_LOG
21.06.24 19:19:45 [INFO] 893 tool used: WOODEN_AXE
21.06.24 19:19:45 [INFO] 894 skipping damage for other reasons
21.06.24 19:19:45 [INFO] 944 InstantRunner: 2b sadness:340/64/79 - DARK_OAK_LOG
21.06.24 19:19:45 [INFO] 944 breaking DARK_OAK_LOG. custom drops: false - roll: 0.7139582917649991
21.06.24 19:19:45 [INFO] 944 mat: DARK_OAK_LOG
21.06.24 19:19:45 [INFO] 944 not auto adding!
21.06.24 19:19:45 [INFO] 944 simply breaking sadness:340/64/79 - DARK_OAK_LOG
21.06.24 19:19:45 [INFO] 944 tool used: WOODEN_AXE
21.06.24 19:19:45 [INFO] 946 skipping damage for other reasons
21.06.24 19:19:45 [INFO] 993 InstantRunner: 2b sadness:339/64/79 - DARK_OAK_LOG
21.06.24 19:19:45 [INFO] 993 breaking DARK_OAK_LOG. custom drops: false - roll: 0.5462396439366864
21.06.24 19:19:45 [INFO] 993 mat: DARK_OAK_LOG
21.06.24 19:19:45 [INFO] 993 not auto adding!
21.06.24 19:19:45 [INFO] 993 simply breaking sadness:339/64/79 - DARK_OAK_LOG
21.06.24 19:19:45 [INFO] 993 tool used: WOODEN_AXE
21.06.24 19:19:45 [INFO] 994 skipping damage for other reasons
21.06.24 19:19:46 [INFO] 43 InstantRunner: 2b sadness:339/68/78 - DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 43 breaking DARK_OAK_LOG. custom drops: false - roll: 0.24393304586501896
21.06.24 19:19:46 [INFO] 43 mat: DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 43 not auto adding!
21.06.24 19:19:46 [INFO] 43 simply breaking sadness:339/68/78 - DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 43 tool used: WOODEN_AXE
21.06.24 19:19:46 [INFO] 45 skipping damage for other reasons
21.06.24 19:19:46 [INFO] 93 InstantRunner: 2b sadness:339/66/79 - DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 93 breaking DARK_OAK_LOG. custom drops: false - roll: 0.624928003924111
21.06.24 19:19:46 [INFO] 93 mat: DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 93 not auto adding!
21.06.24 19:19:46 [INFO] 93 simply breaking sadness:339/66/79 - DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 93 tool used: WOODEN_AXE
21.06.24 19:19:46 [INFO] 95 skipping damage for other reasons
21.06.24 19:19:46 [INFO] 196 InstantRunner: 2b sadness:339/65/78 - DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 197 breaking DARK_OAK_LOG. custom drops: false - roll: 0.8606659630259894
21.06.24 19:19:46 [INFO] 197 mat: DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 197 not auto adding!
21.06.24 19:19:46 [INFO] 197 simply breaking sadness:339/65/78 - DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 197 tool used: WOODEN_AXE
21.06.24 19:19:46 [INFO] 198 skipping damage for other reasons
21.06.24 19:19:46 [INFO] 205 InstantRunner: 2b sadness:340/65/78 - DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 205 breaking DARK_OAK_LOG. custom drops: false - roll: 0.3690538941728966
21.06.24 19:19:46 [INFO] 205 mat: DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 205 not auto adding!
21.06.24 19:19:46 [INFO] 206 simply breaking sadness:340/65/78 - DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 206 tool used: WOODEN_AXE
21.06.24 19:19:46 [INFO] 207 skipping damage for other reasons
21.06.24 19:19:46 [INFO] 247 InstantRunner: 2b sadness:340/68/78 - DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 247 breaking DARK_OAK_LOG. custom drops: false - roll: 0.9909721223638417
21.06.24 19:19:46 [INFO] 247 mat: DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 247 not auto adding!
21.06.24 19:19:46 [INFO] 247 simply breaking sadness:340/68/78 - DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 247 tool used: WOODEN_AXE
21.06.24 19:19:46 [INFO] 248 skipping damage for other reasons
21.06.24 19:19:46 [INFO] 299 InstantRunner: 2b sadness:340/69/78 - DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 299 breaking DARK_OAK_LOG. custom drops: false - roll: 0.35581899319395227
21.06.24 19:19:46 [INFO] 299 mat: DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 299 not auto adding!
21.06.24 19:19:46 [INFO] 299 simply breaking sadness:340/69/78 - DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 299 tool used: WOODEN_AXE
21.06.24 19:19:46 [INFO] 300 skipping damage for other reasons
21.06.24 19:19:46 [INFO] 344 InstantRunner: 2b sadness:339/71/79 - DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 344 breaking DARK_OAK_LOG. custom drops: false - roll: 0.2228830775114342
21.06.24 19:19:46 [INFO] 344 mat: DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 344 not auto adding!
21.06.24 19:19:46 [INFO] 344 simply breaking sadness:339/71/79 - DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 344 tool used: WOODEN_AXE
21.06.24 19:19:46 [INFO] 344 skipping damage for other reasons
21.06.24 19:19:46 [INFO] 393 InstantRunner: 2b sadness:339/71/78 - DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 393 breaking DARK_OAK_LOG. custom drops: false - roll: 0.7440879619423009
21.06.24 19:19:46 [INFO] 393 mat: DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 393 not auto adding!
21.06.24 19:19:46 [INFO] 393 simply breaking sadness:339/71/78 - DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 393 tool used: WOODEN_AXE
21.06.24 19:19:46 [INFO] 394 skipping damage for other reasons
21.06.24 19:19:46 [INFO] 443 InstantRunner: 2b sadness:339/67/78 - DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 443 breaking DARK_OAK_LOG. custom drops: false - roll: 0.5697076460243938
21.06.24 19:19:46 [INFO] 443 mat: DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 443 not auto adding!
21.06.24 19:19:46 [INFO] 443 simply breaking sadness:339/67/78 - DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 443 tool used: WOODEN_AXE
21.06.24 19:19:46 [INFO] 444 skipping damage for other reasons
21.06.24 19:19:46 [INFO] 493 InstantRunner: 2b sadness:340/65/79 - DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 493 breaking DARK_OAK_LOG. custom drops: false - roll: 0.42031735950018656
21.06.24 19:19:46 [INFO] 493 mat: DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 493 not auto adding!
21.06.24 19:19:46 [INFO] 493 simply breaking sadness:340/65/79 - DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 493 tool used: WOODEN_AXE
21.06.24 19:19:46 [INFO] 493 skipping damage for other reasons
21.06.24 19:19:46 [INFO] 543 InstantRunner: 2b sadness:340/66/79 - DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 543 breaking DARK_OAK_LOG. custom drops: false - roll: 0.9587660905239693
21.06.24 19:19:46 [INFO] 543 mat: DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 543 not auto adding!
21.06.24 19:19:46 [INFO] 543 simply breaking sadness:340/66/79 - DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 543 tool used: WOODEN_AXE
21.06.24 19:19:46 [INFO] 544 skipping damage for other reasons
21.06.24 19:19:46 [INFO] 593 InstantRunner: 2b sadness:340/67/79 - DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 593 breaking DARK_OAK_LOG. custom drops: false - roll: 0.3436131405358478
21.06.24 19:19:46 [INFO] 593 mat: DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 593 not auto adding!
21.06.24 19:19:46 [INFO] 593 simply breaking sadness:340/67/79 - DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 593 tool used: WOODEN_AXE
21.06.24 19:19:46 [INFO] 593 skipping damage for other reasons
21.06.24 19:19:46 [INFO] 643 InstantRunner: 2b sadness:340/68/79 - DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 643 breaking DARK_OAK_LOG. custom drops: false - roll: 0.7539555660869957
21.06.24 19:19:46 [INFO] 643 mat: DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 643 not auto adding!
21.06.24 19:19:46 [INFO] 643 simply breaking sadness:340/68/79 - DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 643 tool used: WOODEN_AXE
21.06.24 19:19:46 [INFO] 643 skipping damage for other reasons
21.06.24 19:19:46 [INFO] 694 InstantRunner: 2b sadness:340/69/79 - DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 694 breaking DARK_OAK_LOG. custom drops: false - roll: 0.3221236443279756
21.06.24 19:19:46 [INFO] 694 mat: DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 694 not auto adding!
21.06.24 19:19:46 [INFO] 694 simply breaking sadness:340/69/79 - DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 694 tool used: WOODEN_AXE
21.06.24 19:19:46 [INFO] 694 skipping damage for other reasons
21.06.24 19:19:46 [INFO] 743 InstantRunner: 2b sadness:340/70/79 - DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 743 breaking DARK_OAK_LOG. custom drops: false - roll: 0.12285901218262252
21.06.24 19:19:46 [INFO] 743 mat: DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 743 not auto adding!
21.06.24 19:19:46 [INFO] 743 simply breaking sadness:340/70/79 - DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 743 tool used: WOODEN_AXE
21.06.24 19:19:46 [INFO] 743 skipping damage for other reasons
21.06.24 19:19:46 [INFO] 793 InstantRunner: 2b sadness:340/71/79 - DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 793 breaking DARK_OAK_LOG. custom drops: false - roll: 0.8829128928463124
21.06.24 19:19:46 [INFO] 793 mat: DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 793 not auto adding!
21.06.24 19:19:46 [INFO] 793 simply breaking sadness:340/71/79 - DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 793 tool used: WOODEN_AXE
21.06.24 19:19:46 [INFO] 794 skipping damage for other reasons
21.06.24 19:19:46 [INFO] 843 InstantRunner: 2b sadness:339/69/78 - DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 843 breaking DARK_OAK_LOG. custom drops: false - roll: 0.06996148074798969
21.06.24 19:19:46 [INFO] 843 mat: DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 843 not auto adding!
21.06.24 19:19:46 [INFO] 843 simply breaking sadness:339/69/78 - DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 843 tool used: WOODEN_AXE
21.06.24 19:19:46 [INFO] 844 skipping damage for other reasons
21.06.24 19:19:46 [INFO] 894 InstantRunner: 2b sadness:339/67/79 - DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 894 breaking DARK_OAK_LOG. custom drops: false - roll: 0.11454849209312523
21.06.24 19:19:46 [INFO] 894 mat: DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 894 not auto adding!
21.06.24 19:19:46 [INFO] 894 simply breaking sadness:339/67/79 - DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 894 tool used: WOODEN_AXE
21.06.24 19:19:46 [INFO] 895 skipping damage for other reasons
21.06.24 19:19:46 [INFO] 944 InstantRunner: 2b sadness:339/68/79 - DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 944 breaking DARK_OAK_LOG. custom drops: false - roll: 0.43938912553786647
21.06.24 19:19:46 [INFO] 944 mat: DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 944 not auto adding!
21.06.24 19:19:46 [INFO] 944 simply breaking sadness:339/68/79 - DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 944 tool used: WOODEN_AXE
21.06.24 19:19:46 [INFO] 945 skipping damage for other reasons
21.06.24 19:19:46 [INFO] 993 InstantRunner: 2b sadness:340/66/78 - DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 993 breaking DARK_OAK_LOG. custom drops: false - roll: 0.5554553181452829
21.06.24 19:19:46 [INFO] 993 mat: DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 993 not auto adding!
21.06.24 19:19:46 [INFO] 993 simply breaking sadness:340/66/78 - DARK_OAK_LOG
21.06.24 19:19:46 [INFO] 993 tool used: WOODEN_AXE
21.06.24 19:19:46 [INFO] 993 skipping damage for other reasons
21.06.24 19:19:47 [INFO] 44 InstantRunner: 2b sadness:339/70/78 - DARK_OAK_LOG
21.06.24 19:19:47 [INFO] 44 breaking DARK_OAK_LOG. custom drops: false - roll: 0.01972282898005151
21.06.24 19:19:47 [INFO] 44 mat: DARK_OAK_LOG
21.06.24 19:19:47 [INFO] 44 not auto adding!
21.06.24 19:19:47 [INFO] 44 simply breaking sadness:339/70/78 - DARK_OAK_LOG
21.06.24 19:19:47 [INFO] 44 tool used: WOODEN_AXE
21.06.24 19:19:47 [INFO] 44 skipping damage for other reasons
21.06.24 19:19:47 [INFO] 93 InstantRunner: 2b sadness:339/69/79 - DARK_OAK_LOG
21.06.24 19:19:47 [INFO] 93 breaking DARK_OAK_LOG. custom drops: false - roll: 0.6037816248132565
21.06.24 19:19:47 [INFO] 93 mat: DARK_OAK_LOG
21.06.24 19:19:47 [INFO] 93 not auto adding!
21.06.24 19:19:47 [INFO] 93 simply breaking sadness:339/69/79 - DARK_OAK_LOG
21.06.24 19:19:47 [INFO] 93 tool used: WOODEN_AXE
21.06.24 19:19:47 [INFO] 93 skipping damage for other reasons
21.06.24 19:19:47 [INFO] 143 InstantRunner: 2b sadness:339/70/79 - DARK_OAK_LOG
21.06.24 19:19:47 [INFO] 143 breaking DARK_OAK_LOG. custom drops: false - roll: 0.41231092357082266
21.06.24 19:19:47 [INFO] 143 mat: DARK_OAK_LOG
21.06.24 19:19:47 [INFO] 143 not auto adding!
21.06.24 19:19:47 [INFO] 143 simply breaking sadness:339/70/79 - DARK_OAK_LOG
21.06.24 19:19:47 [INFO] 143 tool used: WOODEN_AXE
21.06.24 19:19:47 [INFO] 144 skipping damage for other reasons
21.06.24 19:19:47 [INFO] 193 InstantRunner: 2b sadness:340/70/78 - DARK_OAK_LOG
21.06.24 19:19:47 [INFO] 193 breaking DARK_OAK_LOG. custom drops: false - roll: 0.6127488047302188
21.06.24 19:19:47 [INFO] 193 mat: DARK_OAK_LOG
21.06.24 19:19:47 [INFO] 193 not auto adding!
21.06.24 19:19:47 [INFO] 193 simply breaking sadness:340/70/78 - DARK_OAK_LOG
21.06.24 19:19:47 [INFO] 193 tool used: WOODEN_AXE
21.06.24 19:19:47 [INFO] 193 skipping damage for other reasons
21.06.24 19:19:47 [INFO] 244 InstantRunner: 2b sadness:340/71/78 - DARK_OAK_LOG
21.06.24 19:19:47 [INFO] 244 breaking DARK_OAK_LOG. custom drops: false - roll: 0.13333088469515875
21.06.24 19:19:47 [INFO] 244 mat: DARK_OAK_LOG
21.06.24 19:19:47 [INFO] 244 not auto adding!
21.06.24 19:19:47 [INFO] 244 simply breaking sadness:340/71/78 - DARK_OAK_LOG
21.06.24 19:19:47 [INFO] 244 tool used: WOODEN_AXE
21.06.24 19:19:47 [INFO] 244 skipping damage for other reasons
21.06.24 19:19:49 [INFO] 696 onEntityChangeBlock : SHEEP
21.06.24 19:19:50 [INFO] 396 onEntityChangeBlock : SHEEP
```

**Comment by molor** [comment]
dark_oak.yml
```yml
Blocks:
  # Here you can add extra blocks that can be expected inside or around tree leaves
  Materials:
  - 'minecraft:dark_oak_leaves'
  Top:
    # Height above the trunk to check for leaves
    Height: 2
  Middle:
    # Radius around the trunk to check for leaves
    Radius: 3
  Custom Drops:
    # Drop chances for extra drops. 1.0 would be 100% chance!
    Items:
      'minecraft:dark_oak_leaves': 0.01
# The parent tree config to inherit from, recursively
Parent: default
# The permission required for this tree type
Permission: treeassist.destroy.darkoak
Replanting:
  # The material to place
  Material: 'minecraft:dark_oak_sapling'
Trunk:
  # How high does it need to be to qualify as a tree
  Minimum Height: 3
  # How thick is the trunk
  Thickness: 4
  # One of these materials needs to be part of the trunk for it to count as a trunk
  Materials:
  - 'minecraft:dark_oak_log'
  # Saplings can be at different height
  Uneven Bottom: true
  # Look for branches
  Branch: false
# Version number for automagical config updates
Version: 7.1013
```

**Comment by slipcor** [comment]
please reset Thickness to 2 - this definitely is a breaking change, and set Branch to true again

I just obliterated a ton of REALLY janky hill-forest trees, not a single block remained unbroken :/ and I set the "leaves: false" as well, this did not do anything to hinder the functionality.

I will continue to obliterate the dark forest next to a hill I am at

**Comment by molor** [comment]
> please reset Thickness to 2 - this definitely is a breaking change, and set Branch to true again

hmm, this is left after my tests, I forgot to change it back /:
https://paste.gg/p/anonymous/6843b3e302fd40389117a345dd619177

**Comment by molor** [comment]
Now it works like this
![image](https://user-images.githubusercontent.com/19989717/123300222-09ce0380-d523-11eb-96af-56169405ade8.png)

**Comment by slipcor** [comment]
can you tell me the seed and coordinates? I just obliterated a forest and not a single log stayed...
![dark_oak](https://user-images.githubusercontent.com/970122/123300517-f1f28180-d51a-11eb-993b-0ed2745d239e.png)

**Comment by molor** [comment]
... this part of tree are always remains for some reason
![image](https://user-images.githubusercontent.com/19989717/123300611-68937d00-d523-11eb-89fe-c965d4c7de28.png)

**Comment by molor** [comment]
I can't give seed because map was lost it (extreme server testing in the past), but cat give u a structure with this tree
[tree1.zip](https://github.com/slipcor/TreeAssist/files/6710839/tree1.zip)

**Comment by slipcor** [comment]
I am very confused as to why this happens on your server. for me it even removes this if it is only one single diagonal block ... or if the trunk has AIR in it even on top. Let me double check my dark_oak.yml is vanilla.

Yep, please make sure that your dark_oak.yml looks like this:

https://github.com/slipcor/TreeAssist/blob/master/src/main/resources/trees/overworld/dark_oak.yml

This works just fine for me. Which exact server version are you running? Maybe this is a server implementation issue? I am running paper

I will try to import this structure

**Comment by slipcor** [comment]
I imported the structure and it got removed absolutely fine... I think I need all your configs - some config setting is going nuts... could it be one in default.yml ?

**Comment by molor** [comment]
> Yep, please make sure that your dark_oak.yml looks like this:

Same file on my side

> I am running paper

Paper 1.16.5 too

I remembered why I set `Trunk.Thickness` to 4!
Load this structure [tree2.zip](https://github.com/slipcor/TreeAssist/files/6710880/tree2.zip) and try to break a tree near sign with yellow text with `Trunk.Thickness` = 2. Nothing happens, this isn't a tree for TreeAssist. Now change `Trunk.Thickness` to 4 and try again — it works, but there are problem that described in 1st post :c

**Comment by molor** [comment]
OK, maybe my configuration is very different from the default, although I don't remember changing anything much

[config.zip](https://github.com/slipcor/TreeAssist/files/6710890/config.zip)

**Comment by slipcor** [comment]
okay it finally did happen with your config. I will make a diff check to see what could be the problem 👍

**Comment by slipcor** [comment]
alright I found out why it is bugging out. 

You added "air" blocks to [default.yml] "Ground Blocks" list - is there a reason for that?

Basically what you did there is potentially make every log that is in the air its own tree. So when the tree determination looked around the trunk of the dark oak tree you started killing, it found a neighbor tree [up in the sky because sky is a valid ground material] and then only cut AROUND that neighbor tree, leaving that outer "tree" intact.

Removing air, cave air and water from the "valid ground blocks" fixed your issue.

Ground blocks are a very important kind of thing. Ground blocks mean that these blocks are supposed to be allowed "under the tree and it still is a valid tree", so basically .. a tree in the air, right? but this can not work as all branches will be little trees over the air :D

For the tall oak trees, please remove "Thickness: 3" -- this setting means it will only ever consider this a valid tree if the trunk is 3x3 - which is why the branches never were removed - it fell back to "regular oak with no branches"

I checked all other configs and I am fairly sure the rest of the settings is okay.

Thanks a lot for your input anyway though, I noticed an issue with the config updating, I will push a fix for this, there will be some config nodes added to configs in the future (stripped logs are valid trunk material, and more things)

**Comment by molor** [comment]
> You added "air" blocks to [default.yml] "Ground Blocks" list - is there a reason for that?

Oh, yes, this allows player to break trees that for some reason already floats in the air too, instead of breaking every log separately..

**Comment by molor** [comment]
Water added because I found that swamp trees sometimes generated on water.

**Comment by slipcor** [comment]
I am afraid that the only fix for this is to place ground below. we can not really allow air in the trunk blocks or in the ground blocks, because then the tree determination goes rogue af

And I get that with water... water MIGHT be okay. but there is just some randomness of the genereation that really throws a wrench in our gears :)

**Comment by molor** [comment]
Why we can't check for ground only once after first log are breaks, not after every block? Or I misunderstood how it works

**Comment by slipcor** [comment]
The function at issue here is there to make sure that we do not cut into neighbor trees, invalidating them by cutting away their logs. We check around for tree logs, and check if they constitute a valid tree (that we should not infringe upon) - your determination change made all branches into their own trees.

Disabling this feature even if only for you, will mean that you will absolutely have a mess in the roofed forest because dark oak trees grow so close to each other that their branches do overlap a lot

**Comment by molor** [comment]
Now I realized what kind of chaos can be created by TA via changing just a couple of lines in a couple of configurations :D

**Comment by slipcor** [comment]
yeah this is the price we pay for a lot of options, some of them can be really tricky to figure out :D

I am sorry that I have no great solution for you. maybe we can more closely investigate when or why this happens and put in some kind of edge case of "allowing one block of air" between log and valid ground? maybe two... but if we allow two again, this will "false positive" for dark oak trees :/

**Comment by molor** [comment]
We just need something that make this tree valid. I think that, because I'm not sure that every player will run around each tree and see if it flies :D
![image](https://user-images.githubusercontent.com/19989717/123321201-54f41080-d53b-11eb-8320-934f5aa35651.png)

**Comment by molor** [comment]
Well, issue from 1st post was fixed — thanks for pointing for my errors. Now the only issue is trees like above (and https://github.com/slipcor/TreeAssist/commit/470ce5c30fbf23e9bf8f83e3f356410b19fc4723#r52618210, maybe)

**Comment by slipcor** [comment]
yeah, I understand the problem now. I think I have an idea but this will lead to saplings getting placed where the tree ends.

"The bottom of the tree" and saplings are linked together pretty much, I need to look at this for a while to implement it properly.

Give me a week and I think I can fix it!

Can you move this over to a new ticket? This image really says it all :D

**Comment by molor** [comment]
Thanks!

**Comment by molor** [comment]
Could you give me a link to build with latest changes (if it exists), please?
